src/Tabs/Tab.tsx

```diff
@@ -4,11 +4,15 @@ import { StyledTab, StyledTabWrapper } from './Tab.styles';
 
 export interface TabProps {
   tabId: string;
+  onSelect?: () => void;
```

Comment:
Looking at https://edozohq.atlassian.net/browse/MAPS-278, looks like this will need to be `onBeforeChange` (or similar naming). We will need the ability to show the modal, and if the user clicks cancel - not change the tab.